controller/src/beerocks/master/db/db.cpp

```diff
@@ -2565,24 +2570,33 @@ bool db::set_client_initial_radio(const sMacAddr &mac, const sMacAddr &initial_r
 {
     auto node = get_node_verify_type(mac, beerocks::TYPE_CLIENT);
     if (!node) {
-        LOG(ERROR) << "client node not found for mac " << mac;
+        LOG(ERROR) << "Client node not found for mac " << mac;
         return false;
     }
 
-    LOG(DEBUG) << "initial_radio = " << initial_radio_mac;
+    LOG(DEBUG) << "initial_radio=" << initial_radio_mac;
+
+    // Since the initial radio is an internal parameter (not configured by the user), its value
+    // is only relevant if the stay_on_initial_radio is set and although we want its value to be
+    // persistent, we don't want it to affect the client's aging.
+    // This means:
+    // 1. We do not update the timestamp when we update only the initial_radio.
+    // 2. We only set the initial_radio if the stay_on_initial_radio is set.
+    if (node->client_stay_on_initial_radio == eTriStateBool::NOT_CONFIGURED) {
+        LOG(WARNING) << "Configuring initial-radio to " << initial_radio_mac
+                     << " aborted: stay-on-initial-radio is not configured yet";
+        return true;
```

Comment:
shouldn't you return false here?

Reply:
I don't think so if we return false it would indicate the function failed,
But since this isn't a failure scenario it should return true

Reply:
we should return false to indicate to the caller of this function that the set did not complete successfully.